src/common/utils/agent-platform.utils.ts

```diff
@@ -0,0 +1,223 @@
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import path from 'path';
+import untildify from 'untildify';
+import AppConfig from '../../app-config/config';
+import { CreatePlatformInput } from '../../architect/platform/platform.utils';
+
+const SERVICE_ACCOUNT_NAME = 'architect-agent';
+
+export class AgentPlatformUtils {
+
+  private static getLocalServerAgentIP(): string {
+    return 'host.docker.internal';
+  }
+
+  private static getLocalServerAgentPort(): string {
+    const container_name_results = execa.sync('docker', ['ps', '-f', 'name=agent-server', '--format', '{{.Names}}']);
+    const results = execa.sync('docker', ['port', container_name_results.stdout, '9081/tcp']);
+    return results.stdout.split(':')[1];
+  }
+
+  public static getServerAgentHost(agent_server_host: string): string {
+    const host = agent_server_host.toLocaleLowerCase().trim();
+    if (host[host.length - 1] === ':') {
+      return `${host}${this.getLocalServerAgentPort()}`;
+    }
+    if (host !== 'local') {
+      return agent_server_host;
+    }
+    return `https://${this.getLocalServerAgentIP()}:${this.getLocalServerAgentPort()}`;
+  }
+
+  public static async configureAgentPlatform(
+    flags: any,
+    description: string,
+  ): Promise<CreatePlatformInput> {
+    const kubeconfig_path = untildify(flags.kubeconfig);
+
+    const set_kubeconfig = ['--kubeconfig', untildify(kubeconfig_path), '--namespace', 'default'];
+
+    let use_existing_sa;
+    try {
+      await execa('kubectl', [
+        ...set_kubeconfig,
+        'get', 'sa', SERVICE_ACCOUNT_NAME,
+        '-o', 'json',
+      ]);
+      use_existing_sa = true;
+    } catch {
+      use_existing_sa = false;
+    }
+
+    // Make sure the existing SA uses cluster-admin role binding
+    if (use_existing_sa) {
+      const { stdout } = await execa('kubectl', [
+        ...set_kubeconfig,
+        'get', 'clusterrolebinding',
+        '-o', 'json',
+      ]);
+      const clusterrolebindings = JSON.parse(stdout);
+      const sa_binding = clusterrolebindings.items.find(
+        (rolebinding: any) =>
+          rolebinding.subjects &&
+          rolebinding.subjects.length > 0 &&
+          rolebinding.subjects.find(
+            (subject: any) =>
+              subject.kind === 'ServiceAccount' &&
+              subject.name === SERVICE_ACCOUNT_NAME
+          )
+      );
+
+      if (!sa_binding) {
+        await execa('kubectl', [
+          ...set_kubeconfig,
+          'create',
+          'clusterrolebinding',
+          `${SERVICE_ACCOUNT_NAME}-cluster-admin`,
+          '--clusterrole',
+          'cluster-admin',
+          '--serviceaccount',
+          `default:${SERVICE_ACCOUNT_NAME}`,
+        ]);
+      }
+    }
+
+    if (!use_existing_sa) {
+      CliUx.ux.action.start('Creating the service account');
+      await AgentPlatformUtils.createKubernetesServiceAccount(untildify(kubeconfig_path), SERVICE_ACCOUNT_NAME);
+      const secret_yml = `
```

Comment:
Might be nice to add this to something like a generic `PlatformUtils` so that this isn't duplicated and we can update both platforms in the future if we need to. There are some other duplicated things in this file that could be added to that generic util file too

Reply:
I debated this but because I plan on removing the old logic this seemed fine and kept everything separate without adding a layer of abstraction that was not needed in the future.